lib/spdy-transport/protocol/http2/parser.js

```diff
@@ -154,9 +154,18 @@ Parser.prototype.onFrameBody = function onFrameBody(header, buffer, callback) {
   else if (header.type === frameType.X_FORWARDED_FOR)
     this.onXForwardedFrame(header, buffer, callback);
   else
-    callback(null, { type: 'unknown: ' + header.type });
+    this.onUnknownFrame(header, buffer, callback);
 };
 
+Parser.prototype.onUnknownFrame = function onUnknownFrame(header, buffer, callback) {
```

Comment:
It is not valid as far as I far as I see it, because of this:

```
A CONTINUATION frame MUST be preceded by a HEADERS, PUSH_PROMISE or
CONTINUATION frame without the END_HEADERS flag set. A recipient that observes
violation of this rule MUST respond with a connection error (Section 5.4.1) of type
PROTOCOL_ERROR.
```

Basically, CONTINUATION frames are required after HEADERS without END_HEADERS, and no other frames are allowed before the CONTINUATION. So I guess it should be safe to disallow all frames except GOAWAY. Do you agree?